### PR TITLE
server: Implement protobuf validation

### PR DESCRIPTION
**Pull request by rmichela**
* Add protoc-gen-validate 0.1.0 java dependencies
* Generate PGV validators as part of the maven build
* Validate received requests and reject them if validation fails

### CODE REVIEW

**Comment by rmichela** [comment]
@snowp @joeyb can you please take a look?

File: server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java [hunk 1]
```diff
@@ -219,10 +224,18 @@ public DiscoveryRequestStreamObserver(String defaultTypeUrl,
...
       this.executor = executor;
+      validator = new ReflectiveValidatorIndex().validatorFor(DiscoveryRequest.class);
     }
 
     @Override
     public void onNext(DiscoveryRequest request) {
+      try {
+        validator.assertValid(request);
+      } catch (ValidationException e) {
+        closeWithError(ValidationExceptions.asStatus(e).asException());
```
Comment (joeyb) [inline server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java]: Would be great to have a negative test case that asserts a sample validation error.

Comment (ramaraochavali) [reply server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java]: One thing to note AFAIK, Envoy's [DiscoveryRequest](https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/discovery.proto#L23) does not have any protobuf validations. So adding this for validating requests may not be of much help?

Comment (rmichela) [reply server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java]: Good point. Should `java-control-plane` validate responses? If so, what should it do? Log the error and continue?

**Comment by joeyb** [comment]
Closing this for now since `DiscoveryRequest` doesn't actually have any validation rules to validate right now.

If validation rules are added and we re-open this, then we also need to consider the behavior when those validation rules fail. As-is, if we just reject the request then Envoy is likely going to just create another identical request and we'll end up in a failure loop.

**Pull request closed without merging**